src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -253,31 +254,42 @@ internal SymbolTreeInfo Create()
                     try
                     {
                         _metadataReader = moduleMetadata.GetMetadataReader();
+
+                        // First, walk all the symbols from metadata, populating the parentToChilren
+                        // map accordingly.
+                        GenerateMetadataNodes();
+
+                        // Now, once we populated the initial map, go and get all the inheritance 
+                        // information for all the types in the metadata.  This may refer to 
+                        // types that we haven't seen yet.  We'll add those types to the parentToChildren
+                        // map accordingly.
+                        PopulateInheritanceMap();
+
+                        // Clear the set of type definitions we read out of this piece of metadata.
+                        _allTypeDefinitions.Clear();
                     }
-                    catch (BadImageFormatException)
+                    catch (BadImageFormatException ex) when (ReportWatson(ex))
                     {
+                        // any operation off metadata can throw BadImageFormatException
```

Comment:
Do we want to report some user-visible diagnostic (to the error list e.g.)?

Reply:
we already did this in other places, and we didn't show anything to users. so unless we want to change that, I will leave that up to @jinujoseph

Reply:
Perhaps you can file a follow up issue to do so, to minimize this change. I think there is value in letting users know things are wrong. I am doing this in the EnC implementation now. It helps us prevent "it doesn't work and I don't know why" kind of bug reports.

Reply:
ya , file a new issue to improvise this
I think we should take this fix for 15.6 and minimize the change

Reply:
@tmat are you creating UI to show it? or info bar or output window? or diagnostics?

Reply:
opened issue - https://github.com/dotnet/roslyn/issues/24459

Reply:
It's a diagnostic that goes to the Error List.